File: update_signals.js
const fs = require('fs');

const file = 'market-intel/data/signals.json';
let data = [];
if (fs.existsSync(file)) {
  try {
    data = JSON.parse(fs.readFileSync(file, 'utf8'));
  } catch (e) {}
}

data.push({
  "timestamp": new Date().toISOString(),
  "signals": [
    {
      "asset": "BTC",
      "signal": "BUY",
      "strength": 0.75,
      "strength_pre_synthesis": 0.85,
      "adjustments": {
        "event_override": -0.15,
        "sentiment_confirmation": 0.05,
        "divergence_penalty": 0,
        "net": -0.10
      },
      "confluence_factors": 3,
      "reasoning": "BTC breaking out. Whale flows: -1,834 BTC (24h) / -8,200 BTC (7d) -> strong accumulation trend. Trends aligned, confidence HIGH (1.0). Bullish divergence: price dropped yet whales accumulating heavily (+0.15 total whale boost). Funding rate -0.015% (shorts paying heavily, +0.10 boost). Combined confluence = HIGH-CONVICTION BUY (0.85). ⚠️ High-impact event risk",
      "entry": 72500,
      "stop": 70000,
      "tp1": 75000,
      "tp2": 78000,
      "tp3": 82000,
      "price_current": 72741,
      "price_24h_change": 1.2
    },
    {
      "asset": "GOLD_FUTURES",
      "signal": "BUY",
      "strength": 0.70,
      "strength_pre_synthesis": 0.82,
      "adjustments": {
        "event_override": -0.15,
        "sentiment_confirmation": 0.03,
        "divergence_penalty": 0,
        "net": -0.12
      },
      "confluence_factors": 2,
      "reasoning": "Gold futures holding strong amidst geopolitical uncertainties and a steady DXY. Real yields remain supportive for safe-haven assets. Trend is upward with solid support at previous breakout levels. ⚠️ High-impact event risk",
      "entry": 2340,
      "stop": 2310,
      "tp1": 2375,
      "tp2": 2400,
      "tp3": 2450,
      "price_current": 2345.5,
      "price_24h_change": 0.8
    }
  ],
  "macro": {
    "risk_sentiment": "RISK_OFF",
    "vix": "Elevated (approx. 20-25)",
    "summary": "Geopolitical tensions and a Middle East energy shock have driven up oil prices, stalling inflation at 2.4%. Consequently, Fed officials have raised their 2026 inflation outlook and are showing growing openness to rate hikes instead of cuts, shifting the narrative toward a 'no-landing' scenario."
  },
  "sentiment": {
    "fear_greed": "FEAR",
    "label": "FEAR",
    "crypto_signal_bias": "BUY",
    "gold_signal_bias": "BUY"
  },
  "delivered": true,
  "agents_used": ["crypto-analyst", "gold-analyst", "macro-scout", "sentiment-radar"]
});

if (data.length > 100) {
  data = data.slice(-100);
}

fs.mkdirSync('market-intel/data', { recursive: true });
fs.writeFileSync(file, JSON.stringify(data, null, 2));
console.log('Updated signals.json');

File: update_signals.py
import json
import os
from datetime import datetime

file_path = 'market-intel/data/signals.json'

data = []
if os.path.exists(file_path):
    with open(file_path, 'r') as f:
        try:
            data = json.load(f)
        except:
            data = []

new_entry = {
  "timestamp": datetime.utcnow().isoformat() + "Z",
  "signals": [
    {
      "asset": "BTC",
      "signal": "SELL",
      "strength": 0.60,
      "strength_pre_synthesis": 0.70,
      "adjustments": {
        "event_override": -0.15,
        "sentiment_confirmation": 0.05,
        "divergence_penalty": 0.00,
        "net": -0.10
      },
      "confluence_factors": 2,
      "reasoning": "Failed to break resistance; macro event override applied.",
      "entry": 71000,
      "stop": 73000,
      "tp1": 68000,
      "tp2": 65000,
      "tp3": 60000,
      "price_current": 71184,
      "price_24h_change": -0.71
    },
    {
      "asset": "ETH",
      "signal": "SELL",
      "strength": 0.55,
      "strength_pre_synthesis": 0.65,
      "adjustments": {
        "event_override": -0.15,
        "sentiment_confirmation": 0.05,
        "divergence_penalty": 0.00,
        "net": -0.10
      },
      "confluence_factors": 2,
      "reasoning": "Following BTC weakness; macro event override applied.",
      "entry": 2200,
      "stop": 2350,
      "tp1": 2000,
      "tp2": 1800,
      "tp3": 1500,
      "price_current": 2180,
      "price_24h_change": -2.97
    },
    {
      "asset": "SOL",
      "signal": "SELL",
      "strength": 0.50,
      "strength_pre_synthesis": 0.60,
      "adjustments": {
        "event_override": -0.15,
        "sentiment_confirmation": 0.05,
        "divergence_penalty": 0.00,
        "net": -0.10
      },
      "confluence_factors": 2,
      "reasoning": "Following broad market downtrend; macro event override applied.",
      "entry": 83,
      "stop": 88,
      "tp1": 75,
      "tp2": 70,
      "tp3": 60,
      "price_current": 82.27,
      "price_24h_change": -2.60
    }
  ],
  "macro": {
    "risk_sentiment": "RISK_OFF",
    "vix": 22.5,
    "summary": "Rates hold steady, inflation concerns persist."
  },
  "sentiment": {
    "fear_greed": 25,
    "label": "FEAR",
    "crypto_signal_bias": "SELL",
    "gold_signal_bias": "HOLD"
  },
  "delivered": True,
  "agents_used": ["crypto-analyst", "gold-analyst", "macro-scout", "sentiment-radar"]
}

data.append(new_entry)
if len(data) > 100:
    data = data[-100:]

with open(file_path, 'w') as f:
    json.dump(data, f, indent=2)

print("done")
